src/main/java/org/ohdsi/circe/cohortdefinition/builders/CriteriaSqlBuilder.java

```diff
@@ -30,25 +30,29 @@ public String getCriteriaSql(T criteria, BuilderOptions options) {
     query = embedJoinClauses(query, joinClauses);
     query = embedWhereClauses(query, whereClauses);
 
+    query = embedAdditionalColumns(query, criteria, options);
+
+    return query;
+  }
+
+  protected String embedAdditionalColumns(String query, T criteria, BuilderOptions options) {
     if (options != null) {
       List<CriteriaColumn> filteredColumns = options.additionalColumns.stream()
-              .filter((column) -> !this.getDefaultColumns().contains(column))
-              .collect(Collectors.toList());
+        .filter((column) -> !this.getDefaultColumns().contains(column))
+        .collect(Collectors.toList());
       if (filteredColumns.size() > 0) {
-        query = StringUtils.replace(query, "@additionalColumns", ", " + this.getAdditionalColumns(filteredColumns));
+        return StringUtils.replace(query, "@additionalColumns", ", " + this.getAdditionalColumns(criteria, filteredColumns));
       } else {
-        query = StringUtils.replace(query, "@additionalColumns", "");
+        return StringUtils.replace(query, "@additionalColumns", "");
       }
     } else {
-      query = StringUtils.replace(query, "@additionalColumns", "");
+      return StringUtils.replace(query, "@additionalColumns", "");
     }
-
-    return query;
   }
 
   protected abstract String getTableColumnForCriteriaColumn(CriteriaColumn column);
 
-  protected String getAdditionalColumns(List<CriteriaColumn> columns) {
+  protected String getAdditionalColumns(T criteria, List<CriteriaColumn> columns) {
```

Comment:
Why was the criteria parameter added to this function when it's not even used in it?